### PR TITLE
Refactor TopologyManagerImport

### PR DESCRIPTION
**Pull request by amichair**
Posting this here for review before pushing, since some of it is a matter of style or preference... I find it much easier to understand now but there may be room for further improvements. What do you think?

### CODE REVIEW

File: topology-manager/src/main/java/org/apache/aries/rsa/topologymanager/importer/TopologyManagerImport.java [hunk 1]
```diff
@@ -105,105 +96,86 @@ public void remove(RemoteServiceAdmin rsa) {
...
+     * i.e. unimports previously imported endpoints that are no longer possible,
+     * and imports new possible endpoints that are not already imported.
+     *
+     * @param filter the filter whose endpoints are synchronized
+     */
     private void synchronizeImports(final String filter) {
         try {
-            unImportForGoneEndpoints(filter);
-            importServices(filter);
+            Set<EndpointDescription> possible = importPossibilities.get(filter);
```
Comment (cschneider) [inline topology-manager/src/main/java/org/apache/aries/rsa/topologymanager/importer/TopologyManagerImport.java]: I like the move to streams but I think I would keep the methods unImportForGoneEndpoints(filter) and       importServices(filter) they nicely split up the method.

**Review by cschneider — APPROVED** [review]

File: topology-manager/src/main/java/org/apache/aries/rsa/topologymanager/importer/TopologyManagerImport.java (hunk 1 above)
Comment (amichair) [reply topology-manager/src/main/java/org/apache/aries/rsa/topologymanager/importer/TopologyManagerImport.java]: Can they be called importAddedEndpoints and unimportRemovedEndpoints or some better name?

Comment (cschneider) [reply topology-manager/src/main/java/org/apache/aries/rsa/topologymanager/importer/TopologyManagerImport.java]: Sure. I did not want to mandate the naming. Just the idea to separate into two methods.

**Comment by amichair** [comment]
Rebased and updated with splitting synchronizeImports as suggested and with the reference null checks from the ARIES-1943 fix.

Note that this does not include the logs that were originally added along with the null checks, as they convolute the code, there is no informative toString in the printed objects, and I'm not sure it's an important case to know about (it's ok if a registration was closed, so as long as there is no NPE bug...). Do you think that's reasonable?

Any other suggestions or improvements?

**Comment by cschneider** [comment]
I think it looks good as it is. We can think of further optimisations in the next step if we want.

**Pull request merged**